src/model/fileinfo.py

```diff
@@ -1,11 +1,59 @@
-# TODO: Create FileInfo class. It should contain the following private variables:
-#  - __prog_path:           str
-#  - __prog_name:           str
-#  - __prog_description:    str
-#  - __args:                List[Argument]
-#  - __is_main_runnable:    bool
-
-# TODO: Create constructor for FileInfo class, that sets all variables above.
-# TODO: Create getters for all the variables in FileInfo class.
-# TODO: Create add_argument(arg: Argument) method.
-# TODO: Create setter function for __is_main_runnable variable.
+from typing import List # for the argument list
+from argument import Argument  # to have access to argument class
+
+"""
+FileInfo class containing the following private variables:
+ - __prog_path:           str
+ - __prog_name:           str
+ - __prog_description:    str
+ - __args:                List[Argument]
+ - __is_main_runnable:    bool
+"""
+class FileInfo:
+    """
+    Constructor for FileInfo class, that sets all variables above.
```

Comment:
In python docstring, developers don't usually write documentation for constructors. The variables usually listed in the class's documentation under
"Attributes::
    __prog_path: str...
"